### PR TITLE
Fix amx gemm scratch size alloc

### PR DESCRIPTION
**Pull request by egeor**

### CODE REVIEW

File: src/generator_gemm_sse_avx_avx2_avx512.c [hunk 1]
```diff
@@ -108,8 +108,10 @@ LIBXSMM_API_INTERN void libxsmm_generator_gemm_sse_avx_avx2_avx512_kernel_wrappe
 LIBXSMM_API_INTERN void libxsmm_generator_gemm_sse_avx_avx2_avx512_kernel( libxsmm_generated_code*        io_generated_code,
                                                                            libxsmm_loop_label_tracker*    io_loop_label_tracker,
                                                                            const libxsmm_gp_reg_mapping*  i_gp_reg_mapping,
-                                                                           const libxsmm_gemm_descriptor* i_xgemm_desc ) {
+                                                                           const libxsmm_gemm_descriptor* i_xgemm_desc_const ) {
   libxsmm_micro_kernel_config l_micro_kernel_config;
+  libxsmm_gemm_descriptor xgemm_desc_mod = *i_xgemm_desc_const;
+  libxsmm_gemm_descriptor *i_xgemm_desc = (libxsmm_gemm_descriptor*) &xgemm_desc_mod;
```
Comment (alheinecke) [inline src/generator_gemm_sse_avx_avx2_avx512.c:114]: nit: local variables should start with l_

File: src/generator_gemm_sse_avx_avx2_avx512.c [hunk 2]
```diff
@@ -108,8 +108,10 @@ LIBXSMM_API_INTERN void libxsmm_generator_gemm_sse_avx_avx2_avx512_kernel_wrappe
 LIBXSMM_API_INTERN void libxsmm_generator_gemm_sse_avx_avx2_avx512_kernel( libxsmm_generated_code*        io_generated_code,
                                                                            libxsmm_loop_label_tracker*    io_loop_label_tracker,
                                                                            const libxsmm_gp_reg_mapping*  i_gp_reg_mapping,
-                                                                           const libxsmm_gemm_descriptor* i_xgemm_desc ) {
+                                                                           const libxsmm_gemm_descriptor* i_xgemm_desc_const ) {
```
Comment (alheinecke) [inline src/generator_gemm_sse_avx_avx2_avx512.c:111]: why renaming?

**Review by alheinecke — CHANGES_REQUESTED** [review]
only a nit

File: src/generator_gemm_sse_avx_avx2_avx512.c (hunk 2 above)
Comment (egeor) [reply src/generator_gemm_sse_avx_avx2_avx512.c:111]: For the purposes of this (to restrict the scratch size when doing c vnni for non-amx and have it independent of ldc) I had to change the descriptor to use everywhere ldc = m for the gemm writing to scratch, and using the actial ldc only in the vnni cvt tpp at the end when storing from the scratch to the C buffer. Thus I had to create a new modifiable local xgemm descriptor (same thing that was done above for the transa  purposes to modify the descriptor)

Comment (alheinecke) [reply src/generator_gemm_sse_avx_avx2_avx512.c:111]: so local,non-const variable with l_ and and keep the signature as it should fix it.

Comment (egeor) [reply src/generator_gemm_sse_avx_avx2_avx512.c:111]: Then we'd have to change every single occurrence of the i_xgemm name that appears throughout this function to l_xgemm ... I am not saying this is not doable, but with the current code we change the name in the signature and nothing else.

If naming convention is more important then I can change it.

Comment (alheinecke) [reply src/generator_gemm_sse_avx_avx2_avx512.c:111]: yes let's do this as I'm cleaning-up in every routine I see it. Just in order to improve quality, in vim "v" "%s/x/y" should do the job in a routine.

**Review by alheinecke — APPROVED** [review]

**Pull request merged**